### ISSUE TITLE
Localization

### ISSUE THREAD

**Issue by olzraiti** [mozilla-services/react-jsonschema-form#205]
Currently there is no way to localize the form (add/remove/submit -buttons, error messages).
Localizations could be in a separate schema (example below). This would allow uniform localization technique for fields provided by the form and also custom schema fields. However, this wouldn't provide a way to localize error messages. Maybe a error message formatter function could be given to the form as a parameter. 

jsonSchema:

``` js
{type: "object",
 properties: {
     custom: {
         type: "object",
         properties: { ... }
     },
     array: {
          type: "array",
          items: { ... }
      }
 }
}
```

uiSchema:

``` js
{
 custom: {
     "ui:field": someField
 }
}

```

localizationschema:

``` js
{
    array: {
        add: "localized add text",
        delete: "localized delete text"
    }, custom: {
        { ... }
    }
}

```

**Comment by vixriihi** [comment]
Or could we add dependency to something like https://github.com/yahoo/react-intl and pass in the translations in the form properties?

**Comment by olzraiti** [comment]
After #285 is merged, the only thing that can't be localized are the jsonschema validations.

The jsonschema validations can be localized by inspecting the ValidationError object that jsonschema.validate() returns (see: https://github.com/tdegrunt/jsonschema/issues/147). In order to allow  this lib's users to localize the jsonschema validations, it should be either possible to disable jsonschema validating so the users can do the validation themself, or provide a validation interceptor that would receive the ValidationError object that jsonschema returns.

**Comment by olzraiti** [comment]
@n1k0 : Would you accept a pull request if I wrote one? I'd like to have some feedback about my previous message, if I start working on this. :-)

**Comment by n1k0** [comment]
@olzraiti sure, let's just polish and land #285 first, see my latest comments on the PR :)

**Comment by DimitryDushkin** [comment]
i18n is quite important. I think react-intl integration would be the simplest.

**Comment by olzraiti** [comment]
Closed by #432

**Comment by afeld** [comment]
cc #739
